common/src/main/java/org/conscrypt/SSLUtils.java

```diff
@@ -48,6 +50,38 @@
             Boolean.parseBoolean(System.getProperty("org.conscrypt.useEngineSocketByDefault"));
 
     /**
+     * This is the maximum overhead when encrypting plaintext as defined by
+     * <a href="https://www.ietf.org/rfc/rfc5246.txt">rfc5264</a>,
+     * <a href="https://www.ietf.org/rfc/rfc5289.txt">rfc5289</a> and openssl implementation itself.
+     *
+     * Please note that we use a padding of 16 here as openssl uses PKC#5 which uses 16 bytes
+     * whilethe spec itself allow up to 255 bytes. 16 bytes is the max for PKC#5 (which handles it
+     * the same way as PKC#7) as we use a block size of 16. See <a
+     * href="https://tools.ietf.org/html/rfc5652#section-6.3">rfc5652#section-6.3</a>.
+     *
+     * 16 (IV) + 48 (MAC) + 1 (Padding_length field) + 15 (Padding) + 1 (ContentType) + 2
+     * (ProtocolVersion) + 2 (Length)
+     *
+     * TODO: We may need to review this calculation once TLS 1.3 becomes available.
+     */
+    private static final int MAX_ENCRYPTION_OVERHEAD_LENGTH = 15 + 48 + 1 + 16 + 1 + 2 + 2;
```

Comment:
This seems like it's delving too deeply into the particular implementation. `SSL3_RT_MAX_ENCRYPTED_OVERHEAD` already defines something from BoringSSL.